pkg/asset/asset.go

```diff
@@ -50,3 +59,35 @@ func PersistToFile(asset WritableAsset, directory string) error {
 	}
 	return nil
 }
+
+// PersistTargets saves the target assets on disk.
+// It also removes any previous generated assets.
```

Comment:
@yifan-gu what do think about consuming the assets from disk when loading them from disk. So `FilePatternFetcher` performs the consuming (reading and removing) the files part?

Reply:
@abhinavdahiya If there's any errors when generating new assets, the old ones will be removed and hard to recover.